frame/support/test/tests/pallet.rs

```diff
@@ -644,29 +631,44 @@ fn inherent_expand() {
 	assert!(inherent.check_extrinsics(&block).fatal_error());
 
 	let block = Block::new(
-		Header::new(
-			1,
-			BlakeTwo256::hash(b"test"),
-			BlakeTwo256::hash(b"test"),
-			BlakeTwo256::hash(b"test"),
-			Digest::default(),
-		),
+		header.clone(),
 		vec![
 			UncheckedExtrinsic { function: Call::Example(pallet::Call::foo(1, 1)), signature: None },
 			UncheckedExtrinsic { function: Call::Example(pallet::Call::foo_transactional(0)), signature: None },
+			UncheckedExtrinsic { function: Call::Example(pallet::Call::foo_transactional(0)), signature: None },
+			UncheckedExtrinsic { function: Call::Example(pallet::Call::foo_transactional(0)), signature: None },
 		],
 	);
 
-	assert!(Runtime::ensure_inherents_are_first(&block).is_ok());
+	assert_eq!(Runtime::ensure_inherents_are_first(&block).ok().unwrap(), Some(1));
 
```

Comment:
```suggestion

	// Signed extrinsic is the first non-inherent
	let block = Block::new(
		header.clone(),
		vec![
			UncheckedExtrinsic { function: Call::Example(pallet::Call::foo(1, 1)), signature: None },
			UncheckedExtrinsic { function: Call::Example(pallet::Call::foo_transactional(0)), signature: Some((1, (), ())) },
			UncheckedExtrinsic { function: Call::Example(pallet::Call::foo_transactional(0)), signature: None },
			UncheckedExtrinsic { function: Call::Example(pallet::Call::foo_transactional(0)), signature: None },
		],
	);

	assert_eq!(Runtime::ensure_inherents_are_first(&block).ok().unwrap(), Some(1));
	
```

Looks like we just are always testing for unsinged extrinsics. Having one test to show it will behave the same with signed extrinsics would catch a regression that the other tests do not appear capable of

Reply:
equivalent test added in 62bdb6808f9cf3d52552a2ff04f87f8255729107